extensions/amp-story/1.0/amp-story-store-service.js

```diff
@@ -215,18 +222,16 @@ const actions = (state, action, data) => {
             [StateProperty.BOOKEND_STATE]: !!data,
             [StateProperty.PAUSED_STATE]: !!data,
           }));
-    case Action.TOGGLE_EMBEDDED_COMPONENT:
+    case Action.TOGGLE_INTERACTIVE_COMPONENT:
+      data = /** @type {InteractiveComponentDef} */ (data);
       return /** @type {!State} */ (Object.assign(
           {}, state, {
```

Comment:
As seen offline, because the new state is an object, you might want to add a comparison key, which would ensure your subscribers would only get called when the state got changed